TACC/Core-Portal
# FP-1082/FP-1083: Fix Color Contrast of Accent Color & Body Text

## To Do: ##

<details>
<summary>✓ Blocking</summary>

- [x] [check sidebar black on purple contrast](https://github.com/TACC/Core-Portal/pull/452#issuecomment-907328704)
- [x] 113c82ff, d4abad4, dc876a5, aea3bfd solve sidebar black on purple contrast
- [x] [check whether app tray and sidebar should use same background color](https://github.com/TACC/Core-Portal/pull/452#discussion_r697576970)
- [x] 04d1a10 [add mapping of out-of-date dev and design colors to standard color variables in `colors.css`](https://github.com/TACC/Core-Portal/pull/452#discussion_r697567444)

</details>

<details>
<summary>✓ Non-Blocking</summary>

- [x] [ask design whether app tray and sidebar should use same background color](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1630603710012300)
- [x] [FP-1192](https://jira.tacc.utexas.edu/browse/FP-1192) create ticket to update design doc colors
- [x] [propose to design to use bold sidebar text to distinguish hover and active texts](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1630604234014700)
- [x] [FP-1191](https://jira.tacc.utexas.edu/browse/FP-1191) prepare task to update code based on decisions from design

</details>



## Overview: ##

Fix most color contrast failures for link and body text.



## Related Jira tickets: ##

* __[FP-1082](https://jira.tacc.utexas.edu/browse/FP-1082) (implement changes in code)__
* [FP-1083](https://jira.tacc.utexas.edu/browse/FP-1083) (propose colors to design, receive approval)



## Summary of Changes: ##

- Darken the body text color.
- Fix value and usage of accent colors.
- Use the accent color variables.
- Rename & Remove some accent color variables.
- Change normal and weak accent colors (for accessible text contrast).
- Explain absence of #E6E0FB color (that is used in design doc).



## Testing Steps: ##

1. Review pages and see that body text (not buttons nor links) is darker than before:
    - [`#707070`](https://hslpicker.com/#707070) "dark" → [`#484848`](https://hslpicker.com/#484848) "x-dark"
    - [`#484848`](https://hslpicker.com/#484848) "x-dark" → [`#222222`](https://hslpicker.com/#222222) "xx-dark"
2. Review pages and see that purple colors (links, badges, etc.) are "stronger" than before:
	- [`#9D85EF`](https://hslpicker.com/#9D85EF) → [`#7754E8`](https://hslpicker.com/#7754E8) "normal"
	- [`#9D85EF20`](https://hslpicker.com/#9D85EF,0.12) → [`#7754E840`](https://hslpicker.com/#7754E8,0.25) "weak" (looks like `#DFD7FA` on white bkdg.) \*
	- [`#DFDAF5`](https://hslpicker.com/#DFDAF5) → [`#7754E840`](https://hslpicker.com/#7754E8,0.25) "weak" (looks like `#DFD7FA` on white bkdg.) \*
    - [`#D2CCE7`](https://hslpicker.com/#D2CCE7) (var name change only, now "alt") (used for notification, see "Notes" 2)

> \* [How do you know?](https://filosophy.org/code/online-tool-to-lighten-color-without-alpha-channel/)



## UI Photos:

<details><summary>Body Text</summary>

| Description | Image |
| --- | --- |
| Dashboard | <img height="200px" title="Dashboard" src="https://user-images.githubusercontent.com/62723358/129248184-3bb944c5-4738-42d5-93d0-1649d0064920.png" /> |
| Data Files | <img height="200px" title="Data Files" src="https://user-images.githubusercontent.com/62723358/129248188-bdcc1291-12c7-47cd-b3a4-0a9f2a57dc57.png" /> |
| header, menu | <img width="1280" alt="header, menu" src="https://user-images.githubusercontent.com/62723358/129248190-229b0dba-5994-4b2c-8189-c63d203d30ab.png"> |
| information modal | <img height="200px" title="information modal" src="https://user-images.githubusercontent.com/62723358/129248193-b059678a-122e-42f2-8459-76cb2123e799.png" /> |
| My Account | <img height="200px" title="My Account" src="https://user-images.githubusercontent.com/62723358/129248195-41a2539b-4ca8-4fed-a179-d7ff8a20f2fb.png" /> |
| Shared Workspaces | <img height="200px" alt="Shared Workspaces" src="https://user-images.githubusercontent.com/62723358/130840020-cb87cdad-c841-4b2e-b4fb-7e4a43fff01d.png"> |
| Site Search | <img height="200px" title="Site Search" src="https://user-images.githubusercontent.com/62723358/129248200-ef817870-0e6c-41ac-a043-17ebb91a5104.png" /> |
| Job History Modal | <img height="200px" alt="Job History Modal" src="https://user-images.githubusercontent.com/62723358/130840107-9b7b215d-e2d9-4443-90a8-322b98f9b035.png"> |

</details>



<details><summary>Accent Color</summary>

| Description | Image |
| --- | --- |
| Allocations, Button | <img height="200px" title="Allocations, Button" src="https://user-images.githubusercontent.com/62723358/127596646-54fc05c9-6d1f-4643-afa5-fb72dd52c36e.png" /> |
| Applications, Message Link | <img height="200px" title="Applications, Message Link" src="https://user-images.githubusercontent.com/62723358/127596648-e1ceab26-4900-483d-9204-e026e299a194.png" /> |
| UI Patterns, Pagination | <img height="200px" title="UI Patterns, Pagination" src="https://user-images.githubusercontent.com/62723358/127596652-9385ad69-96e9-4add-b005-e763690f32aa.png" /> |
| History, Disabled Link | <img height="200px" title="History, Disabled Link" src="https://user-images.githubusercontent.com/62723358/127596651-8c7d0429-bff2-47ba-8276-7f8d455c2434.png" /> |

</details>



<details><summary>Link Active/Hover States</summary>

| Test | Sidebar | App Tray | App Sidebar | Data Files | Search | Allocs | History |
| --- | --- | --- | --- | --- | --- | --- | --- |
| ![Sidebar Color Contrast Test](https://user-images.githubusercontent.com/62723358/131886115-3e0d1b75-887d-4ea5-8f5f-92de6e46f9a0.png) | ![Sidebar Color Contrast Fix](https://user-images.githubusercontent.com/62723358/131889255-ee00342b-1090-4997-99a6-1a23f02d08de.png) | ![App Sidebar Color Contrast Fix](https://user-images.githubusercontent.com/62723358/131886117-7614c483-de42-45db-a489-856e5395b426.png) | ![App Tray Color Contrast Fix](https://user-images.githubusercontent.com/62723358/131886112-68303687-26f9-45e2-950a-299cc4843a88.png) | ![Data Files Sidebar Color Contrast](https://user-images.githubusercontent.com/62723358/131886996-52e497fb-c2b0-4e0d-b75a-086aa1605f6f.png) | ![Search Sidebar Color Contrast](https://user-images.githubusercontent.com/62723358/131887000-d9cdd994-cc21-438a-a814-d2815d3de3b4.png) | ![FP-1083 Allocs Hover-Active](https://user-images.githubusercontent.com/62723358/132908287-31a885c2-0625-448b-830f-db5b29d602c6.png) | ![FP-1083 History Active](https://user-images.githubusercontent.com/62723358/132908306-95690a89-8d39-46bd-b702-4064e5021dff.png) |

</details>



## Notes: ##

1. `<Message>` color contrast changes (other types/colors) are delayed until [FP-1145](https://jira.tacc.utexas.edu/browse/FP-1145).
2. `<NotifcationToast>` color contrast changes are delayed until [FP-1149](https://jira.tacc.utexas.edu/browse/FP-1149).

<details><summary>To Do (Done)</summary>

- [x] c766551 Ensure Shared Workspaces description text is darker.
- [x] 380910c Fix bugs found during new tasks.
- [x] 592ad5a Notify developers (in code) that colors differ from design.
- [x] [FP-1175](https://jira.tacc.utexas.edu/browse/FP-1175) Create ticket for pill color contrast.\*
- [x] ([explanation](https://github.com/TACC/Core-Portal/pull/452/files#r692543234)) Confirm that disabled badge is "not a thing".†
- [x] Update screenshots out-of-date since 380910c and later commits.

> \* Hint: Colors should be as saturated and and bright as the accent color. (I confirmed with designer M.S..)
> † A disabled badge would just be absent.

</details>
